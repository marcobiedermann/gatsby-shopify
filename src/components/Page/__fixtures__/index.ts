import { PageProps } from '../index';

const page: PageProps = {
  body: '',
  title: 'Page Title',
};

export default page;
