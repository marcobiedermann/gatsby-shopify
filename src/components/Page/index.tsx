import React, { FC } from "react"

export interface PageProps {
  body: string
  title: string
}

const Page: FC<PageProps> = props => {
  const { body, title } = props

  return (
    <>
      <h1>{title}</h1>
      <p dangerouslySetInnerHTML={{ __html: body }} />
    </>
  )
}

export default Page
