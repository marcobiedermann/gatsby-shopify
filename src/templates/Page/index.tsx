import React, { FC } from "react"
import Layout from "../../components/Layout"
import Page, { PageProps } from "../../components/Page"

export interface PageTemplateProps {
  pageContext: {
    page: PageProps
  }
}

const PageTemplate: FC<PageTemplateProps> = props => {
  const { pageContext } = props
  const { page } = pageContext

  return (
    <Layout>
      <Page {...page} />
    </Layout>
  )
}

export default PageTemplate
