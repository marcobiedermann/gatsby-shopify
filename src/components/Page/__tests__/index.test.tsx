import { render } from '@testing-library/react';
import React from 'react';
import Page from '..';
import page from '../__fixtures__';

describe('page component', () => {
  it('should render correctly', () => {
    const { container } = render(<Page {...page} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
