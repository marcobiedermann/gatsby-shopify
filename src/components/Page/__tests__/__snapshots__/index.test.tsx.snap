// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page component should render correctly 1`] = `
<h1>
  Page Title
</h1>
`;
